lua/arrow/buffer_persist.lua

```diff
@@ -133,16 +133,24 @@ end
 
 function M.update(bufnr)
 	bufnr = bufnr or vim.api.nvim_get_current_buf()
+	local line_count = vim.api.nvim_buf_line_count(bufnr)
 	local extmarks = vim.api.nvim_buf_get_extmarks(bufnr, ns, { 0, 0 }, { -1, -1 }, {})
+	local deleted = {}
 	if M.local_bookmarks[bufnr] ~= nil then
-		for _, mark in ipairs(M.local_bookmarks[bufnr]) do
+		for index, mark in ipairs(M.local_bookmarks[bufnr]) do
 			for _, extmark in ipairs(extmarks) do
 				local extmark_id, extmark_row, _ = unpack(extmark)
+				if mark.ext_id == extmark_id and line_count < extmark_row + 1 then
+					table.insert(deleted, index)
+				end
 				if mark.ext_id == extmark_id and mark.line ~= extmark_row + 1 then
 					mark.line = extmark_row + 1
 				end
 			end
 		end
+		for _, index in ipairs(deleted) do
+			M.local_bookmarks[bufnr][index] = nil
```

Comment:
this will leave a void in the middle of array, use the lua/arrow/buffer_persist.lua:110 (M.remove)

Reply:
Hmm, the index will change after remove element, I have changed to use tbl_filter